source/docs/software/basic-programming/git-getting-started.rst

```diff
@@ -0,0 +1,250 @@
+
+- Seperate testing environments into *branches*
+- Ability to navigate to a particular *commit* without removing history
+- Ability to manage *commits* in various ways, including combining them
+- Various other features, see `here <https://git-scm.com/about>`__
+
+Prerequisites
+-------------
+
+You have to download and install Git from the following links:
+
+- `Windows <https://git-scm.com/download/win>`_
+- `Mac OS X <https://git-scm.com/download/mac>`_
+- `Linux <https://git-scm.com/download/linux>`_
+
+.. note:: You may need to add Git to your path
+
+Git Vocabulary
+--------------
+
+Git is composed of several primary elements:
+
+- *Repository:* the actual local of your code, including a ``.git`` folder in the root directory
+- *Commits:* a particular saved state of the repository, this includes all files and additions
+- *Branches:* a means of separating various commits, having a unique history. This is primarily used for separating development and stable branches.
+- *Push:* update the remote repository with your local changes
+- *Pull:* update your local repository with the remote changes
+- *Cloning:* retrieving a local copy of a repository to modify
+- *Forking:* duplicating a pre-existing repository to modify, and to compare against the original
+- *Merging:* combining various changes from different branches/commits/forks into a single history
+
+Repository
+----------
+
+A Git repository is a data structure containing the structure, history, and files of a project.
+
+Git repositories consist of:
+
+- A ``.git`` folder. This folder contains the various information about the repository. 
+- A ``.gitignore`` file. This file contains the files or directories that you do *not* want included when you commit.
+- Files and folders. This is the main content of the repository.
+
+Creating the repository
+^^^^^^^^^^^^^^^^^^^^^^^
+
+You can store the repository locally, or through the cloud. `Github <https://github.com/>`_ is a popular free hosting service. Numerous developers use it, and that's what this tutorial will use.
+
+Creating a Github Account
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Go ahead and create a Github account by visiting the `website <https://github.com>`_ and following the own screen prompts.
+
+.. image:: images/image1.png
+
+Local Creation
+~~~~~~~~~~~~~~
+
+After creating and verifying your account, you'll want to visit the homepage. It'll look similar to the shown image.
+
+.. image:: images/image2.png
+
+Click the plus icon in the top right.
+
+.. image:: images/image3.png
+
+Then click *"New Repository"*
+
+.. image:: images/image4.png
+
+Fill out the appropriate information, and then click *"Create repository"*
+
+.. image:: images/image5.png
+
+You should see a screen similar to this
+
+.. image:: images/image6.png
+
+Now you'll want to open a powershell window and navigate to your project directory. An excellent tutorial can be found `here <https://programminghistorian.org/en/lessons/intro-to-powershell>`__
```

Comment:
What if I’m not on windows?

Reply:
We should focus our instructions on windows, as that is by far our most used platform, and most linux/mac users know the terminal. We can have a bit of guide on the other terminals, but we should focus mainly on windows.

Also, we should in here mention `ctrl + ~` in VS Code opens an integrated terminal. Since most teams are VS Code, I don't mind having a bit of platform specifics in there, and its super useful.

Reply:
I added a clarification.